ndc-lambda-sdk/src/schema.ts

```diff
@@ -271,6 +280,42 @@ function convertTypeReferenceToSdkType(typeRef: TypeReference): sdk.Type {
   }
 }
 
+function convertBuiltInScalarTypeIntoSdkSchemaType(typeName: BuiltInScalarTypeName): sdk.ScalarType {
+  switch (typeName) {
+    case BuiltInScalarTypeName.String: return {
+      representation: { type: "string" },
+      aggregate_functions: {},
+      comparison_operators: { "_eq": { type: "equal" } },
+    };
+    case BuiltInScalarTypeName.Float: return {
+      representation: { type: "float64" },
+      aggregate_functions: {},
+      comparison_operators: { "_eq": { type: "equal" } },
+    };
+    case BuiltInScalarTypeName.Boolean: return {
+      representation: { type: "boolean" },
+      aggregate_functions: {},
+      comparison_operators: { "_eq": { type: "equal" } },
+    };
+    case BuiltInScalarTypeName.BigInt: return {
+      representation: { type: "int64" }, // NDC doesn't have a good representation for this type as at v0.1.2, so this is the best representation in the meantime
+      aggregate_functions: {},
+      comparison_operators: { "_eq": { type: "equal" } },
+    };
+    case BuiltInScalarTypeName.DateTime: return {
+      representation: { type: "timestamp" },
+      aggregate_functions: {},
+      comparison_operators: { "_eq": { type: "equal" } },
+    };
+    case BuiltInScalarTypeName.JSON: return {
+      representation: { type: "json" },
+      aggregate_functions: {},
+      comparison_operators: {},
+    };
+    default: return unreachable(typeName);
+  }
+}
+
```

Comment:
Since we don't expose any collections and don't use predicates, comparison operators are kinda unused. I've added equals only because I figure if we're allowing remote joins on these types (even if remote joins in the functions sense means just argument binding), it sort of stands to reason these types are equals-able. Strictly I don't think this is necessary to define explicitly like this, but it doesn't hurt.